packages/block-library/src/rss/index.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Server-side rendering of the `core/rss` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the rss error message.
+ *
+ * @param string $message RSS error message.
+ * @return string|void Void if renders on frontend.
+ */
+function render_rss_error_message( $message ) {
+	return '<div class="components-placeholder"><div class="notice notice-alt notice-error">' . $message . '</div></div>';
+}
+
+/**
+ * Renders the `core/rss` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the block content with received rss items.
+ */
+function render_block_core_rss( $attributes ) {
+
+	$url = $attributes['feedURL'];
+
+	if ( empty( $url ) ) {
+		return;
+	}
+
+	while ( stristr( $url, 'http' ) !== $url ) {
+		$url = substr( $url, 1 );
+	}
+
+	// self-url destruction sequence.
+	if ( in_array( untrailingslashit( $url ), array( site_url(), home_url() ) ) ) {
+		return render_rss_error_message( __( 'Use \'Latest Posts\' block for this domain', 'gutenberg' ) );
```

Comment:
That is correct, yes. No text domain for that.

Also, I‘d probably use sprintf for the block name here and use double quotes for the string in order to avoid backslashes for the single quotes in it.